auslib/db.py

```diff
@@ -977,11 +984,20 @@ def select(self, where=None, transaction=None, **kwargs):
 
     def insert(self, changed_by, transaction=None, dryrun=False, **columns):
         base_columns, condition_columns = self._splitColumns(columns)
-
+        if "change_type" not in base_columns:
+            raise ValueError("Change type is required")
+
+        if base_columns["change_type"] == "delete":
+            for pk in self.base_primary_key:
+                if pk not in base_columns:
+                    raise ValueError("Missing primary key column %s. PK values needed for deletion" % (pk))
+                if base_columns[pk] is None:
+                    raise ValueError("%s value found to be None. PK value can not be None for deletion" % (pk))
```

Comment:
I think these checks need to happen in update() as well - we wouldn't want an existing scheduled deletion to have its PK removed. If you can, moving these to validate() is probably the best thing to do.